### PR TITLE
Temporal: Add Temporal calendar and timezone case insensitive tests

### PR DESCRIPTION
**Pull request by Aditi-1400**
This PR adds tests for the normative change: https://github.com/tc39/proposal-temporal/pull/2395

### CODE REVIEW

**Review by ptomato — CHANGES_REQUESTED** [review]
Looks good (I reviewed a representative subset of the tests) but I think we need to have the argument be some mixed-case form of "iso8601" or "UTC" in all cases. Implementations aren't required to support "japanese" and "Etc/GMT+1", so those tests might fail. So, I'd recommend testing only the ISO calendar and UTC time zone, and then adding just a few representative tests in the `intl402/` directory (maybe just for `new TimeZone`, `TimeZone.from`, `new Calendar`, `Calendar.from`) for mixed-case versions of other time zones and calendars.

File: test/built-ins/Temporal/Now/zonedDateTimeISO/timezone-case-insensitive.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.now.prototype.zoneddatetimeiso
```
Comment (ptomato) [inline test/built-ins/Temporal/Now/zonedDateTimeISO/timezone-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.now.zoneddatetimeiso
```

File: test/built-ins/Temporal/PlainDate/calendar-case-insensitive.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindate.prototype.constructor
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainDate/calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plaindate
```

File: test/built-ins/Temporal/PlainDateTime/calendar-case-insensitive.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.constructor
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainDateTime/calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plaindatetime
```

File: test/built-ins/Temporal/PlainDateTime/from/argument-propertybag-calendar-case-insensitive.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plaindatetime.prototype.from
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainDateTime/from/argument-propertybag-calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plaindatetime.from
```

File: test/built-ins/Temporal/PlainMonthDay/calendar-case-insensitive.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plainmonthday.prototype.constructor
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainMonthDay/calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plainmonthday
```

File: test/built-ins/Temporal/PlainMonthDay/from/argument-propertybag-calendar-case-insensitive.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plainmonthday.prototype.from
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainMonthDay/from/argument-propertybag-calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plainmonthday.from
```

File: test/built-ins/Temporal/PlainYearMonth/calendar-case-insensitive.js [hunk 7]
```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plainyearmonth.prototype.constructor
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainYearMonth/calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plainyearmonth
```

File: test/built-ins/Temporal/PlainYearMonth/from/argument-propertybag-calendar-case-insensitive.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.plainyearmonth.prototype.from
```
Comment (ptomato) [inline test/built-ins/Temporal/PlainYearMonth/from/argument-propertybag-calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.plainyearmonth.from
```

File: test/built-ins/Temporal/TimeZone/from/timezone-case-insensitive.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.timezone.prototype.from
```
Comment (ptomato) [inline test/built-ins/Temporal/TimeZone/from/timezone-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.timezone.from
```

File: test/built-ins/Temporal/ZonedDateTime/calendar-case-insensitive.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.zoneddatetime.prototype.constructor
```
Comment (ptomato) [inline test/built-ins/Temporal/ZonedDateTime/calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.zoneddatetime
```

File: test/built-ins/Temporal/ZonedDateTime/from/argument-propertybag-calendar-case-insensitive.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.zoneddatetime.prototype.from
```
Comment (ptomato) [inline test/built-ins/Temporal/ZonedDateTime/from/argument-propertybag-calendar-case-insensitive.js:5]:
```suggestion
esid: sec-temporal.zoneddatetime.from
```

**Review by ptomato — APPROVED** [review]
Looks good now, thanks for updating!

I noticed something with the ESIDs that I missed on the first review, sorry about that. (the ESID should be equal to the part of the spec text URL after `#`, e.g. for the [Temporal.Calendar constructor](https://tc39.es/proposal-temporal/#sec-temporal.calendar) it's `sec-temporal.calendar`.) I think I've identified all the places where this is an issue, so we can just add the suggestions before merging.

I'll mark it now as "awaiting consensus" so we can merge it after the plenary.

**Comment by Aditi-1400** [comment]
Oops! 🙈 Thanks for all the suggestions! :)

**Review by ptomato — APPROVED** [review]
The normative change that this tests, achieved consensus at the September 2022 TC39 meeting.

**Pull request merged**